recycler/src/main/java/com/elpassion/android/commons/recycler/strategies/items/impl/MapItemsStrategy.kt

```diff
@@ -0,0 +1,72 @@
+package com.elpassion.android.commons.recycler.strategies.items.impl
+
+import com.elpassion.android.commons.recycler.strategies.items.MutableSectionedItemsStrategy
+import com.elpassion.android.commons.recycler.strategies.items.SectionedItemsStrategy
+import java.util.*
+
+class MapItemsStrategy<SectionType, ItemType>(list: Map<SectionType, List<ItemType>>) : SectionedItemsStrategy<SectionType, ItemType> {
+
+    private val items: Map<SectionType, List<ItemType>> = LinkedHashMap<SectionType, List<ItemType>>(list.size).apply {
+        putAll(list.mapValues { ArrayList(it.value) })
+    }
+
+    override fun allItems() = items.values.flatten()
+
+    override fun getSection(section: SectionType) = items[section]!!.toList()
+
+    override fun getRelativePosition(itemPosition: Int) = getRelativePosition(this, items, itemPosition)
+
+    override fun getSectionForItemPosition(itemPosition: Int) = getSectionForItemAdapter(allItems()[itemPosition], items)
+}
+
+class MutableMapItemsStrategy<SectionType, ItemType> : MutableSectionedItemsStrategy<SectionType, ItemType> {
+
+    private val items: MutableMap<SectionType, MutableList<ItemType>>
+
+    constructor() {
+        items = mutableMapOf()
+    }
+
+    constructor(list: Map<SectionType, List<ItemType>>) {
+        items = LinkedHashMap<SectionType, MutableList<ItemType>>(list.size).apply {
+            putAll(list.mapValues { it.value.toMutableList() })
+        }
+    }
+
+
+    override fun addAll(section: SectionType, from: List<ItemType>) {
+        items[section]!!.addAll(from)
+    }
+
+    override fun add(section: SectionType, item: ItemType) {
+        items[section]!!.add(item)
+    }
+
+    override fun clear() {
+        items.clear()
+    }
+
+    override fun set(section: SectionType, from: List<ItemType>) {
+        items.put(section, from.toMutableList())
+    }
+
+    override fun allItems() = items.values.flatten()
+
+    override fun getSection(section: SectionType) = items[section]!!.toList()
+
+    override fun getRelativePosition(itemPosition: Int) = getRelativePosition(this, items, itemPosition)
+
+    override fun getSectionForItemPosition(itemPosition: Int) = getSectionForItemAdapter(allItems()[itemPosition], items)
+
+}
+
+private fun <ItemType, SectionType> getSectionForItemAdapter(itemAdapter: ItemType, adapters: Map<SectionType, List<ItemType>>): SectionType
+        = adapters.filterValues { it.contains(itemAdapter) }.keys.first()
```

Comment:
Looks highly inefficient. I think it can trigger a garbage collector a lot during recycler view scrolling - causing animation interruptions. Or.. I don't know - I haven't seen any actual use case for it all :)